src/DataSource.ts

```diff
@@ -70,21 +70,27 @@ export class DataSource extends DataSourceApi<MyQuery> {
       params: { action: 'get_combined_graph_identifications' },
       data: buildRequestBody(combinedDesc({ host: { host: 'ARANDOMNAME' } })),
       context: {},
-    }).then((response) => {
-      if (
-        response.data.result_code === 1 &&
-        response.data.result === 'Checkmk exception: Currently not supported with this Checkmk Edition' &&
-        get(this, 'instanceSettings.jsonData.edition', 'CEE') === 'CEE'
-      ) {
-        throw new Error('Mismatch between selected checkmk edition and monitoring site edition');
-      }
-
-      return {
-        status: 'success',
-        message: 'Data source is working',
-        title: 'Success',
-      };
-    });
+    })
+      .catch((error) => {
+        let firstLineOfError = error.message.split('\n')[0];
+        if (firstLineOfError === 'Checkmk exception: Currently not supported with this Checkmk Edition') {
+          if (get(this, 'instanceSettings.jsonData.edition', 'CEE') === 'CEE') {
```

Comment:
This could be replaced by a .? chain.

Would be better for typing.

Reply:
had to add some other typing stuff to make this work.